locations/spiders/deichmann.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+import scrapy
+import json
+import base64
+from locations.items import GeojsonPointItem
+
+class Deichmann(scrapy.Spider):
+    name = "deichmann"
+    allowed_domains = ["stores.deichmann.com","stores.dosenbach.ch"]
+    start_urls = (
+        'https://stores.deichmann.com/index.html',
+    )
+
+    def parse(self, response):
+        # base url of all countries
+        countries = response.xpath('//ul[@class="countrydropdown"]/li/a/@href').extract()
+
+        for country in countries:
+            yield scrapy.Request(country, callback=self.parse_country)
+
+    def parse_country(self, response):
+        # parse country per region
+        cities = response.xpath('//a[@class="region-list__link"]/@href').extract()
+
+        for city in cities:
+            yield scrapy.Request(response.urljoin(city), callback=self.parse_city)
+
+    def parse_city(self, response):
+        # parse cities in region
+        stores = response.xpath('//a[@class="button-link"]/@href').extract()
+
+        for store in stores:
+            yield scrapy.Request(response.urljoin(store), callback=self.parse_locations)
+
+    def parse_locations(self, response):
+        # get json which provides all data
+        data = response.xpath('//script[@type="application/ld+json"]/text()').extract_first()
+
+        if data and self.isValidJson(data):
```

Comment:
Instead of parsing and throwing away the result to check for valid JSON, just parse it here, catch the Exception and `return` instead of letting it keep going.